### PR TITLE
Fix TestScoredHostsRandSelectByScore NDF

### PR DESCRIPTION
**Pull request by peterjan**

### CODE REVIEW

File: worker/worker.go [hunk 1]
```diff
@@ -1480,7 +1480,7 @@ func discardTxnOnErr(ctx context.Context, bus Bus, l *zap.SugaredLogger, txn typ
 
 	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
 	if dErr := bus.WalletDiscard(ctx, txn); dErr != nil {
-		l.Errorf("%w: %v, failed to discard txn: %v", *err, errContext, dErr)
+		l.Errorf("%v: %s, failed to discard txn: %v", *err, errContext, dErr)
```
Comment (ChrisSchinnerl) [inline worker/worker.go:1483]: What does this fix?

Comment (peterjan) [reply worker/worker.go:1483]: This prints something like: `2024-06-03T14:59:53+02:00       ERROR   e2e/cluster_test.go:45  %!w(*errors.errorString=&{foo}): bar context, failed to discard txn: baz` - can't find the exact error so had to quickly reproduce in a test.

**Review by ChrisSchinnerl — APPROVED** [review]

File: autopilot/contractor/hosts_test.go [hunk 2]
```diff
@@ -52,20 +52,27 @@ func TestScoredHostsRandSelectByScore(t *testing.T) {
...
 			counts[0]++
 		} else {
 			counts[1]++
 		}
 	}
-	if diff := absDiffInt(counts[0], counts[1]); diff > 40 {
-		t.Fatal("unexpected", counts[0], counts[1], diff)
+	var chi2 float64
+	for i := 0; i < 2; i++ {
+		chi2 += math.Pow(float64(counts[i])-nRuns/2, 2) / nRuns / 2
```
Comment (ChrisSchinnerl) [inline autopilot/contractor/hosts_test.go:73]: If the expected value is `nRuns/2` shouldn't this be `math.Pow(float64(counts[i])-nRuns/2, 2) / (nRuns / 2)`?
Should make a difference whether you divided by `10000` and then `2` or `5000`

**Review by ChrisSchinnerl — CHANGES_REQUESTED** [review]

File: autopilot/contractor/hosts_test.go (hunk 2 above)
Comment (peterjan) [reply autopilot/contractor/hosts_test.go:73]: lol thanks, I blame the chad

**Review by ChrisSchinnerl — APPROVED** [review]

**Pull request merged**